File: Android/app/src/main/kotlin/com/zhechengqi/tollcat/setup/SetupVerifyOutcome.kt
package com.zhechengqi.tollcat.setup

import com.zhechengqi.tollcat.GuideErrorCase

sealed class SetupVerifyOutcome {
    data class Success(val title: String, val detail: String?) : SetupVerifyOutcome()
    data object EmptyReading : SetupVerifyOutcome()
    data class Http(val errorCase: GuideErrorCase) : SetupVerifyOutcome()
    data object Network : SetupVerifyOutcome()
    data object Unknown : SetupVerifyOutcome()

    val isSuccess: Boolean get() = this is Success

    /** 401 / 403 / 空读数需要回教程改权限；断网只是等一会儿再试。 */
    val offersRevisePermissions: Boolean
        get() = when (this) {
            EmptyReading -> true
            is Http -> errorCase.httpStatus == 401 || errorCase.httpStatus == 403
            else -> false
        }
}

File: Packages/MeterKit/Sources/MeterFeatures/Setup/SetupVerifyOutcome.swift
import MeterCore
import MeterPersistence
import MeterProviders

enum SetupVerifyOutcome: Equatable {
    case success(title: String, detail: String)
    case emptyReading
    case http(ErrorCase)
    case network
    case unknown

    /// 401 / 403 / 空读数 / 未知失败需要回教程改权限；断网只是等一会儿再试。
    var offersRevisePermissions: Bool {
        switch self {
        case .http, .emptyReading, .unknown:
            return true
        case .success, .network:
            return false
        }
    }
}
